From 2bd5f0022dde2d15e0544fc15d975a66d3a710cc Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibochen@cloudera.com>
Date: Wed, 17 Aug 2016 13:53:46 -0700
Subject: [PATCH 1772/2848] Revert "HADOOP-10748. HttpServer2 should not load
 JspServlet. Contributed by Haohui Mai."

This reverts commit f2b7a6597c96c90ec28f05e053e82a9dac0f0898.

Change-Id: I8448537eac8e62d034e0e6ce8d9c9125075aebb6
---
 .../java/org/apache/hadoop/http/HttpServer2.java   |   12 ------------
 1 file changed, 12 deletions(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/http/HttpServer2.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/http/HttpServer2.java
index 69a9a11..6426894 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/http/HttpServer2.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/http/HttpServer2.java
@@ -45,7 +45,6 @@
 import javax.servlet.http.HttpServletRequestWrapper;
 import javax.servlet.http.HttpServletResponse;
 
-import com.google.common.collect.ImmutableMap;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.apache.hadoop.HadoopIllegalArgumentException;
@@ -458,17 +457,6 @@ private void addManagedListener(Connector connector) {
   private static WebAppContext createWebAppContext(String name,
       Configuration conf, AccessControlList adminsAcl, final String appDir) {
     WebAppContext ctx = new WebAppContext();
-    ctx.setDefaultsDescriptor(null);
-    ServletHolder holder = new ServletHolder(new DefaultServlet());
-    Map<String, String> params = ImmutableMap. <String, String> builder()
-            .put("acceptRanges", "true")
-            .put("dirAllowed", "false")
-            .put("gzip", "true")
-            .put("useFileMappedBuffer", "true")
-            .build();
-    holder.setInitParameters(params);
-    ctx.setWelcomeFiles(new String[] {"index.html"});
-    ctx.addServlet(holder, "/");
     ctx.setDisplayName(name);
     ctx.setContextPath("/");
     ctx.setWar(appDir + "/" + name);
-- 
1.7.9.5

